### PR TITLE
feat(docs): add karma-setup for typescript

### PR DESCRIPTION
**Pull request by codef0rmer**
With respect to https://github.com/rails/webpacker/issues/838 and probably https://github.com/rails/webpacker/issues/838#issuecomment-331938771 too.

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -1,5 +1,106 @@
...
+
+```js
+// package.json
+"scripts": {
+  "test": "NODE_ENV=test karma start"
+},
+"dependencies": {
+  "typescript": "^2.5.2",
+  "ts-loader": "^2.3.7",
+  "karma": "^1.7.1",
```
Comment (gauravtiwari) [inline docs/testing.md]: `karma` and family should be under devDependencies, no?

Comment (codef0rmer) [reply docs/testing.md]: Yes. Done.

File: docs/testing.md [hunk 2]
```diff
@@ -1,5 +1,106 @@
...
+Webpacker does not setup `Karma` by default, so you've to manually install it along with its dependencies as per your need. Following things marked as optional can be used to fancify the test results (Recommended).
+
+```js
+// package.json
+"scripts": {
+  "test": "NODE_ENV=test karma start"
+},
+"devDependencies": {
+  "typescript": "^2.5.2",
+  "ts-loader": "^2.3.7",
```
Comment (gauravtiwari) [inline docs/testing.md]: Thanks for the other fix. Sorry, I wasn't very clear. I meant, `typescript` and `ts-loader` is a dependency (`dependencies`), where `karma` and other testing utils is for development only (`devDependencies`). Right?

Comment (codef0rmer) [reply docs/testing.md]: Yes. Moved them under dependencies.

**Comment by gauravtiwari** [comment]
thanks 🍰

**Comment by codef0rmer** [comment]
Nom Nom Nom 😄

**Pull request merged**